jreader/json_unmarshal_comparative_benchmark_test.go

```diff
@@ -146,3 +147,25 @@ func benchmarkReadObjectJSONUnmarshal(b *testing.B) {
 		}
 	}
 }
+
+func benchmarkReadArrayOfObjectsJSONUnmarshal(b *testing.B) {
```

Comment:
This is tangential to the current PR, although it does relate to the API for reading JSON objects. I'm just adding a benchmark for parsing a big list of structs, as a slightly better simulation of the larger data structures that the SDK is likely to parse; the existing benchmark of a single struct is so small that it's harder to compare.